server/org.wso2.developerstudio.eclipse.carbonserver44/src/org/wso2/developerstudio/eclipse/carbonserver44/internal/CarbonServerBehavior44.java

```diff
@@ -214,100 +220,91 @@ protected String getAxis2FilePath() {
+			} else {
+				port = getPortfromTransportXML(CarbonServerConstants.HTTPS_NAME_TAG);
 			}
 			ports.add(port);
-			inputSource =  new InputSource(new FileInputStream(xmlDocument));
-			evaluate = (String)catelineXPath.evaluate("/Server/Service/Connector[1]/@port", catelinaDocument, XPathConstants.STRING);
+			inputSource = new InputSource(new FileInputStream(xmlDocument));
+			evaluate = (String) catelineXPath.evaluate(
+					CarbonServerConstants.SERVER_SERVICE_CONNECTOR_XPATH,
+					catelinaDocument, XPathConstants.STRING);
 
-//			xPathExpression=xPath.compile("/:Server/:Ports/:ServletTransports/:HTTP");
-//			evaluate = xPathExpression.evaluate(inputSource);
-			if(!evaluate.equals("")){
-				port = Integer.parseInt(evaluate)+offSet;
-			}else{
-				port = getPortfromTransportXML("http");
+			if (!evaluate.equals(EMPTY_STRING)) {
+				port = Integer.parseInt(evaluate) + offSet;
+			} else {
+				port = getPortfromTransportXML(CarbonServerConstants.HTTP_NAME_TAG);
 			}
 			ports.add(port);
-			
-		} catch (NumberFormatException e) {
-			e.printStackTrace();
-		} catch (FileNotFoundException e) {
-			e.printStackTrace();
-		} catch (XPathExpressionException e) {
-			e.printStackTrace();
-		} catch (ParserConfigurationException e) {
-			e.printStackTrace();
-		} catch (SAXException e) {
-			e.printStackTrace();
-		} catch (IOException e) {
-			e.printStackTrace();
+
+		} catch (NumberFormatException | XPathExpressionException
+				| ParserConfigurationException | SAXException | IOException e) {
+			log.warn("Error occured while adding server transport ports", e);
 		}
 	}
-	
-	private int getPortfromTransportXML(String protocolType){
+
+	private int getPortfromTransportXML(String protocolType) {
 		int port = 0;
 		String transportsXmlPath = getTransportXmlFilePath();
 		XPathFactory factory = XPathFactory.newInstance();
 		File xmlDocument = new File(transportsXmlPath);
-    	try {
-			InputSource inputSource =  new InputSource(new FileInputStream(xmlDocument));
-			XPath xPath=factory.newXPath();
-			XPathExpression  xPathExpression=xPath.compile("/transports/transport[@name='" + protocolType + "']/parameter[@name='port']");
+		try {
+			InputSource inputSource = new InputSource(new FileInputStream(
+					xmlDocument));
+			XPath xPath = factory.newXPath();
+			XPathExpression xPathExpression = xPath
+					.compile("/transports/transport[@name='" + protocolType
+							+ "']/parameter[@name='port']");
 			String evaluate = xPathExpression.evaluate(inputSource);
 			port = Integer.parseInt(evaluate);
-		} catch (NumberFormatException e) {
-			// TODO Auto-generated catch block
-			e.printStackTrace();
-		} catch (FileNotFoundException e) {
-			// TODO Auto-generated catch block
-			e.printStackTrace();
-		} catch (XPathExpressionException e) {
-			// TODO Auto-generated catch block
-			e.printStackTrace();
+		} catch (NumberFormatException | FileNotFoundException
+				| XPathExpressionException e) {
```

Comment:
There is no core Java api to avoid this exception. So this exception will throw if invalid argument passed for Integer.parseInt(). 
Here we need to handle it because otherwise  it will escape the block and throw up through the stack. It is not the behavior expected in this method. Because if this method throws this exception it will try to get ports using another method.